src/lens_odm_parser/core.clj

```diff
@@ -167,48 +167,74 @@
   (->> (conform-lc-kw x)
        (s/conform :odm/tx-type)))
 
-(defn- tag= [tag]
-  #(= tag (:tag %)))
-
-(defn- check-tag [tag e]
-  (when-not (= tag (:tag e))
-    (throw
-      (ex-info
-        (format "Expected tag %s but got %s." (name tag) (name (:tag e)))
-        {:type ::validation-error :element e :expected-tag tag}))))
-
-(defn- check-sub-tag [tag e]
-  (when-not (some (tag= tag) (:content e))
-    (throw
-      (ex-info
-        (format "Expected to find element with tag %s as child of element with tag %s."
-                (name tag) (name (:tag e)))
-        {:type ::validation-error :element e :expected-child-tag tag}))))
-
-(defmulti parse*
-  "Parses XML Element. Dispatches by :tag."
+(defn- coerce-oid [s]
+  (coerce :odm.data-formats/oid s))
+
+(defn- coerce-integer [s]
+  (coerce :odm.xml/integer s))
+
+(defn- coerce-float [s]
+  (coerce :odm.xml/float s))
+
+(defmulti parse-elem*
+  "Parses a single XML Element. Dispatches by :tag. Should return a map with a
+  single key-value pair which will be merged together with the maps from other
+  parses. Multiple values with the same key are merged with `into`."
   {:arglists '([element])}
   :tag)
 
-(s/fdef parse
+(s/fdef parse-elem
   :args (s/cat :element ::element))
 
-(defn parse [element]
-  (parse* element))
+(defn parse-elem [element]
+  (parse-elem* element))
+
+(defmethod parse-elem* :default [_])
+
+(defn parse-children
+  "Parses every child using `parse` and merges the results. Keys occuring more
+   than once will be merged using `into`."
+  [content]
+  (apply (partial merge-with into) (map parse-elem content)))
+
+(defmulti parse-attr*
+  "Parses XML Attribute. Dispatches by attribute name or a combination of root
```

Comment:
Dispatches on a vector of element tag and attribute name with a fallback on attribute name only.